### PR TITLE
DE34274: Hide the advanced search link with a dom-if since hidden wil…

### PR DESCRIPTION
**Pull request by njostonehouse**
…l be overridden by class styles.

### CODE REVIEW

File: src/d2l-all-courses.js [hunk 1]
```diff
@@ -101,9 +101,11 @@ $_documentContainer.innerHTML = `<dom-module id="d2l-all-courses">
 							</d2l-dropdown>
 						</div>
 					</div>
-					<div class="search-and-filter-row advanced-search-link" hidden$="[[!_showAdvancedSearchLink]]">
```
Comment (njostonehouse) [inline src/d2l-all-courses.js]: `hidden` combined with `class` styles ends up showing the link regardless of the `_showAdvancedSearchLink` value. Changing it to `dom-if` hides the advanced search link when it isn't specified.

**Comment by njostonehouse** [comment]
Well that's a relevant test failure. Hmm.

File: src/d2l-all-courses.js (hunk 1 above)
Comment (m6jones) [reply src/d2l-all-courses.js]: We could use `.search-and-filter-row[hidden] { display: none; }` to make hidden work again

Comment (njostonehouse) [reply src/d2l-all-courses.js]: Aha! I like that a lot better. I'll give it a shot.

Comment (edwardg8) [reply src/d2l-all-courses.js]: that's indeed much cleaner change

**Pull request merged**